_api-reference/ingest-apis/pipeline-failures.md

```diff
@@ -0,0 +1,140 @@
+    {
+      "date": {
+        "field": "timestamp_field",
+        "target_field": "timestamp",
+        "formats": ["yyyy-MM-dd HH:mm:ss"],
+   "on_failure": [
+    {
+      "set": {
+        "field": "ingest_error",
+        "value": "failed"
+      }
+    }
+  ]
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+If the processor fails, OpenSearch logs the failure and continues to run all remaining processors in the search pipeline. To check whether there were any failures, you can use [ingest pipeline metrics]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/pipeline-failures/#ingest-pipeline-metrics).
+{: tip}
+
+## Ingest pipeline metrics
+
+To view ingest pipeline metrics, use the [Nodes Stats API]({{site.url}}{{site.baseurl}}/api-reference/nodes-apis/nodes-stats/):
+
+```json
+GET /_nodes/stats/ingest?filter_path=nodes.*.ingest
+```
+{% include copy-curl.html %}
+
+The response contains statistics for all ingest pipelines, for example:
+
+```json
+ {
+  "nodes": {
+    "iFPgpdjPQ-uzTdyPLwQVnQ": {
+      "ingest": {
+        "total": {
+          "count": 28,
+          "time_in_millis": 82,
+          "current": 0,
+          "failed": 9
+        },
+        "pipelines": {
+          "user-behavior": {
+            "count": 5,
+            "time_in_millis": 0,
+            "current": 0,
+            "failed": 0,
+            "processors": [
+              {
+                "append": {
+                  "type": "append",
+                  "stats": {
+                    "count": 5,
+                    "time_in_millis": 0,
+                    "current": 0,
+                    "failed": 0
+                  }
+                }
+              }
+            ]
+          },
+           "remove_ip": {
+            "count": 5,
+            "time_in_millis": 9,
+            "current": 0,
+            "failed": 2,
+            "processors": [
+              {
+                "remove": {
+                  "type": "remove",
+                  "stats": {
+                    "count": 5,
+                    "time_in_millis": 8,
+                    "current": 0,
+                    "failed": 2
+                  }
+                }
+              }
+            ]
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+## Troubleshooting failures
+
+The following are tips on troubleshooting ingest pipeline failures:
+
+1. Check the logs: OpenSeach logs contain information about the ingest pipeline that failed, including the processor that failed and the reason for the failure.
+2. Inspect the document: If the ingest pipeline failed, then the document that was being processed will be in its respective index. 
+3. Check the processor configuration: It is possible the processor configuration is incorrect. To check this you can look at the processor configuration in the JSON object.
+4. Try a different processor: You can try using a different processor. Some processors are better at handling certain types of data than others.
```

Comment:
I think that out of these 4 options, I only find #1 actionable. The other ones are too generic... Maybe it's me though.